src/pages/experiments/[experimentId]/plots-and-tables/volcano/index.jsx

```diff
@@ -54,22 +55,27 @@ const VolcanoPlot = ({ experimentId }) => {
     spec: null,
     maxNegativeLogpValue: null,
   });
-
   useEffect(() => {
-    dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
-  }, []);
-
-  useEffect(() => {
-    // Sync plot and settings with last used config
+    if (!config) {
+      dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+    }
     setComparisonGroup({
       ...plotCellSets,
       type: plotComparisonType,
     });
-
-    // Show plot using last shown data
-    if (data.length === 0) return;
-    setDataPointStatus();
   }, []);
+
+  useEffect(() => {
+    if (!config || !expressionData.length) return;
+    setPlotData(calculateVolcanoDataPoints(config, expressionData));
+  }, [config, expressionData]);
+
+  useEffect(() => {
+    if (plotData.length === 0) return;
+    const generatedSpec = generateSpec(config, plotData);
+    setSpec(generatedSpec);
+  }, [plotData]);
```

Comment:
Can you combine these 2 useEffects into one? `calculateVolcanoDataPoints` calculates the `plotData` right, so why not just store this value into a variable, and input this value directly into `generateSpec` instead of using another `plotData` state?